xmodule/library_content_block.py

```diff
@@ -315,16 +320,20 @@ def _v2_update_children_upstream_version(self):
         Update the upstream and upstream version fields of all children to point to library v2 version of the legacy
         library blocks. This essentially converts this legacy block to new ItemBankBlock.
         """
-        from cms.djangoapps.modulestore_migrator.api import get_target_block_usage_keys
-        blocks = get_target_block_usage_keys(self.source_library_key)
+        from cms.djangoapps.modulestore_migrator import api as migrator_api
         store = modulestore()
         with store.bulk_operations(self.course_id):
-            for child in self.get_children():
-                source_key, _ = self.runtime.modulestore.get_block_original_usage(child.usage_key)
-                child.upstream = str(blocks.get(source_key, ""))
-                # Since after migration, the component in library is in draft state, we want to make sure that sync icon
-                # appears when it is published
-                child.upstream_version = 0
+            children = self.get_children()
+            child_migrations = migrator_api.forward_blocks([child.usage_key for child in children])
```

Comment:
We now forward using mappings from the forwarded migration rather than any old mappings.